File: src/main/java/com/sasfmlzr/apivk/bot/UserBot.java
package com.sasfmlzr.apivk.bot;

import com.sasfmlzr.apiVK.bot.AbstractBot;
import com.sasfmlzr.apiVK.thread.ThreadUserBot;
import com.vk.api.sdk.client.VkApiClient;
import com.vk.api.sdk.client.actors.UserActor;
import com.vk.api.sdk.exceptions.ApiException;
import com.vk.api.sdk.exceptions.ClientException;
import com.vk.api.sdk.objects.users.UserXtrCounters;
import com.vk.api.sdk.queries.users.UserField;
import javafx.embed.swing.SwingFXUtils;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.IOException;
import java.net.URL;

public class UserBot extends AbstractBot {
    private UserActor actor;
    private ThreadUserBot threadUserBot;

    public UserBot(VkApiClient vkApiClient, UserActor actor) {
        this.actor = actor;
        vk = vkApiClient;
        threadUserBot = new ThreadUserBot(botApiClient(), actor);
        try {
            UserXtrCounters botSelfInfo = vk.users().get(actor).fields(UserField.PHOTO_200).execute().get(0);
            userID = botSelfInfo.getId();
            botName = botSelfInfo.getFirstName() + " " + botSelfInfo.getLastName();
            BufferedImage daffyDuckImage = ImageIO.read(new URL(botSelfInfo.getPhoto200()));
            setBotImage(SwingFXUtils.toFXImage(daffyDuckImage, null));
        } catch (ApiException | ClientException | IOException e) {
            e.printStackTrace();
        }
    }

    public void run() {
        Runnable r = threadUserBot;
        new Thread(r).start();
    }

    public UserActor getActor() {
        return actor;
    }

    public void setActor(UserActor actor) {
        this.actor = actor;
    }
}

File: src/main/java/com/sasfmlzr/apivk/bot/UserBot.kt
package com.sasfmlzr.apivk.bot

import com.sasfmlzr.apivk.thread.ThreadUserBot
import com.vk.api.sdk.client.VkApiClient
import com.vk.api.sdk.client.actors.UserActor
import com.vk.api.sdk.objects.users.Fields
import javafx.embed.swing.SwingFXUtils
import javax.imageio.ImageIO

class UserBot(vk: VkApiClient, actor: UserActor) : AbstractBot(vk, actor) {
    private val threadUserBot: ThreadUserBot = ThreadUserBot(botApiClient(), actor)

    init {
        try {
            val botSelfInfo = vk.users().get(actor).fields(Fields.PHOTO_200).execute()[0]
            userID = botSelfInfo.id
            botName = "${botSelfInfo.firstName} ${botSelfInfo.lastName}"
            val daffyDuckImage = ImageIO.read(botSelfInfo.photo200)
            botImage = SwingFXUtils.toFXImage(daffyDuckImage, null)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun run() {
        val r = threadUserBot
        Thread(r).start()
    }
}
